samples/DotNet/Rbac/EventHubsSenderReceiverRbac/Program.cs

```diff
@@ -0,0 +1,193 @@
+                    ServiceAudience.EventHubsAudience
+                    ),
+                TransportType = TransportType.Amqp
+            };
+
+            SendReceive(messagingFactorySettings);
+        }
+
+        static void ClientCredentialsCertScenario()
+        {
+            ClientCredential clientCredential = new ClientCredential(ClientId, ConfigurationManager.AppSettings["clientSecret"]);
+            MessagingFactorySettings messagingFactorySettings = new MessagingFactorySettings
+            {
+                TokenProvider = TokenProvider.CreateAadTokenProvider(
+                    new AuthenticationContext($"https://login.windows.net/{TenantId}"),
+                    clientCredential,
+                    ServiceAudience.EventHubsAudience
+                ),
+                TransportType = TransportType.Amqp
+            };
+
+            SendReceive(messagingFactorySettings);
+        }
+
+        static X509Certificate2 GetCertificate()
+        {
+            List<StoreLocation> locations = new List<StoreLocation>
+                {
+                    StoreLocation.CurrentUser,
+                    StoreLocation.LocalMachine
+                };
+
+            foreach (var location in locations)
+            {
+                X509Store store = new X509Store(StoreName.My, location);
+                try
+                {
+                    store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+                    X509Certificate2Collection certificates = store.Certificates.Find(
+                        X509FindType.FindByThumbprint, ConfigurationManager.AppSettings["thumbPrint"], true);
+                    if (certificates.Count >= 1)
+                    {
+                        return certificates[0];
+                    }
+                }
+                finally
+                {
+                    store.Close();
+                }
+            }
+
+            throw new ArgumentException($"A Certificate with Thumbprint '{ConfigurationManager.AppSettings["thumbPrint"]}' could not be located.");
+        }
+
+        static void SendReceive(MessagingFactorySettings messagingFactorySettings)
+        {
+            MessagingFactory messagingFactory = MessagingFactory.Create($"sb://{EventHubNamespace}/",
+                messagingFactorySettings);
+
+            EventHubClient ehClient = messagingFactory.CreateEventHubClient(EventHubName);
+            ehClient.Send(new EventData(Encoding.UTF8.GetBytes($"{DateTime.UtcNow}")));
+
+            EventHubConsumerGroup consumerGroup = ehClient.GetDefaultConsumerGroup();
+
+            string[] partitionIds = { "0", "1" };
+            Parallel.ForEach(partitionIds, partitionId =>
+            {
+                EventHubReceiver receiver = consumerGroup.CreateReceiver(partitionId);
+                EventData data = receiver.Receive();
+                Console.WriteLine(Encoding.UTF8.GetString(data.GetBytes()));
+                receiver.Close();
```

Comment:
This will always receive the first message in the stream. When client run multiple times receiving the same first message can be confusing. Start receivers from the end of the stream before sending the message. This could be the cleanest solution.